packages/terra-date-picker/src/DateUtil.jsx

```diff
@@ -142,6 +143,103 @@ class DateUtil {
     return undefined;
   }
 
+  /**
+   * Gets the date format order from moment's long date format.
+   * @param {string} momentDateFormat - Moment's long date format.
+   * @return {string} - The date format order. Possible Values (MDY, DMY, YMD)
+   */
+  static getDateFormatOrder(momentDateFormat) {
+    let dateFormatOrder;
+    if (momentDateFormat === 'MM/DD/YYYY') {
+      dateFormatOrder = DateUtil.dateOrder.MDY;
+    } else if (momentDateFormat === 'DD/MM/YYYY') {
+      dateFormatOrder = DateUtil.dateOrder.DMY;
+    } else if (momentDateFormat === 'YYYY/MM/DD') {
+      dateFormatOrder = DateUtil.dateOrder.YMD;
+    }
+
+    return dateFormatOrder;
+  }
+
+  /**
+   * Returns an object consisting of date input values based on the date format.
+   * @param {string} dateOrder - String containing the date input order. Possible values (MDY, DMY, YMD)
+   * @param {string} value - The date/placeholder string.
+   * @param {string} separator - The date separator.
+   * @return {Object} - The object containing the day, month and year values.
+   */
+  static getDateInputValues(dateOrder, value, separator) {
+    let day = '';
+    let month = '';
+    let year = '';
+
+    if (value) {
+      const dateInputParts = value.split(separator);
+
+      switch (dateOrder) {
+        case DateUtil.dateOrder.DMY:
+          [day, month, year] = dateInputParts;
+          break;
+        case DateUtil.dateOrder.MDY:
+          [month, day, year] = dateInputParts;
+          break;
+        case DateUtil.dateOrder.YMD:
+          [year, month, day] = dateInputParts;
+          break;
+        default:
+          [day, month, year] = ['', '', ''];
+      }
+    }
+
+    return { day, month, year };
+  }
+
+  /**
+   * Returns component layout based on locale format.
+   * @param {string} dateOrder - The date input order based on the locale date format.
+   * @param {object} separator - The separator component with the locale appropriate separator.
+   * @param {object} day - The date day component.
+   * @param {object} month - The date month component.
+   * @param {object} year - The date year component.
+   * @return {object} - The object containing the date input components in the order as per the locale date format.
+   */
+  static getInputLayout(dateOrder, separator, day, month, year) {
+    let formatOrder;
+    if (dateOrder === DateUtil.dateOrder.DMY) {
+      formatOrder = (
+        <>
+          {day}
+          {separator}
+          {month}
+          {separator}
+          {year}
+        </>
+      );
+    } else if (dateOrder === DateUtil.dateOrder.MDY) {
+      formatOrder = (
+        <>
+          {month}
+          {separator}
+          {day}
+          {separator}
+          {year}
+        </>
+      );
+    } else {
```

Comment:
We should be more explicit to check for a dateOrder === YMD instead of defaulting if it is neither of the two checks above. There could be custom translations/locales or we may add additional locales in the future so we don't want to default those to YMD. We would need to address those as they come. It would be safer to fail to render the layout instead of incorrectly building it in the wrong format as that could lead to misinterpreting the month with day and that would be more serious.

Reply:
Makes sense.
I have changed it here : https://github.com/cerner/terra-framework/pull/1208/commits/ef6e9b4856db01f837879b86710bbe32a85e6a66